### ISSUE TITLE
TRSVGPath has to be properly commented

### ISSUE THREAD

**Issue by bergel** [moosetechnology/Moose#1139]
From Peter U.:
I'm looking at the implementation of TRSVGPath and I am quite confused

[7:15] 
what are polygons?

[7:16] 
`computeRectangle`, `computePolygons`, `correctCurvesAndPolygonsToZero`, `separateSubPaths`

[7:16] 
these methods I don't get at all… why are they there? what do they do?

[7:17] 
if I have a path, then path is just lines and curves, no?

**Comment by akevalion** [comment]
I have updated the comment section in TRSVGShape, let me know if all of this is clear now.

---Q1------
for interactions with mouse, whe use TRShape>> #includesPoint:

There is not an easy way to say when a TRSVGPath includes a point, the current solution is, after we created the curves we use polygons with AthensPolygonTester.

We do not take bezier lines in that solution.

---Q2------
for that reason we use
* computeRectangle. Calculate the encompassing rectangle with this polygons,
* computePolygons. Create the polygons from the curves
* correctCurvesAndPolygonsToZero. Move  the curves coordinates and polygons in the center of the shape, like all the shapes in roassal.
* separateSubPaths. A SVG can be a set of subpaths, example the SVG of UnitedStates with Alaska, and Hawaii as an entire country

.---Q3------
when you draw it yes, TRSVG needs more information like the  encompassing rectangle, or when this shapes includes a point
